### PR TITLE
Update EC2MetadataUtilServer to only accept localhost address

### PR DESCRIPTION
**Pull request by zoewangg**
Update EC2MetadataUtilServer to only accept localhost address

### CODE REVIEW

File: core/src/test/java/software/amazon/awssdk/core/util/EC2MetadataUtilsServer.java [hunk 1]
```diff
@@ -33,9 +33,9 @@
 
     private ServerSocket server;
 
-    public EC2MetadataUtilsServer(String address, int port)
+    public EC2MetadataUtilsServer(int port)
             throws UnknownHostException, IOException {
-        server = new ServerSocket(port, 1, InetAddress.getByName(address));
+        server = new ServerSocket(port, 1, InetAddress.getByName("localhost"));
```
Comment (dagnir) [inline core/src/test/java/software/amazon/awssdk/core/util/EC2MetadataUtilsServer.java]: should we connect directly to `127.0.0.1` instead so there's no lookup?

Comment (zoewangg) [reply core/src/test/java/software/amazon/awssdk/core/util/EC2MetadataUtilsServer.java]: yup! updated

Comment (dagnir) [reply core/src/test/java/software/amazon/awssdk/core/util/EC2MetadataUtilsServer.java]: Apprently `getLocalHost()` is actually kinda misleading and doesn't return the address of `localhost`: https://stackoverflow.com/a/18488982

Looking at it again, I think what you had originally is fine.

Comment (zoewangg) [reply core/src/test/java/software/amazon/awssdk/core/util/EC2MetadataUtilsServer.java]: yeah, actually, `InetAddress.getLoopbackAddress()` is what we want.

Comment (dagnir) [reply core/src/test/java/software/amazon/awssdk/core/util/EC2MetadataUtilsServer.java]: 👍 nice

**Pull request merged**